src/routes/+layout.svelte

```diff
@@ -358,8 +376,12 @@
                         {/each}
                     </div>
 
-                    <div class="mt-[2.8125vw] flex flex-col items-center gap-[1.5vw]">
-                        {#each Object.entries(icon_mapping.middle) as item}
+                    <div class="relative mt-[2.8125vw] flex flex-col items-center gap-[1.5vw]">
+                        <active-background
+                            class="absolute h-[3.375vw] w-[3.375vw] rounded-[0.5vw] bg-secondary-100 transition-transform duration-300 before:absolute before:-left-[0.12vw] before:top-[1.25vw] before:z-10 before:h-[0.875vw] before:w-[0.25vw] before:rounded-lg before:bg-primary-500"
+                            style="transform: translateY({active_button_index * 4.875}vw);"
```

Comment:
This is the worst type of code i have seen lol

Reply:
Okay I'm doing some research on it